### ISSUE TITLE
Input event is not triggered on image insert

### ISSUE THREAD

**Issue by Dmatafonov** [orthes/medium-editor-insert-plugin#167]
I save post html on 'input' event. But when image is inserted, this event is not triggered.
How should I know that image is inserted to html? Is it a bug or i should listen to some other event?

Input event is recommended by mediumEditor.

```javascript
$(document).on('input', '.blog-input', function (e) {

    });
```

**Comment by orthes** [comment]
```input``` should be triggered on image upload: https://github.com/orthes/medium-editor-insert-plugin/blob/master/src/js/images.js#L293 . When I try it, it works.

```input``` event is triggered on successful upload only. If your server didn't return 200 status, ```uploadDone()``` function is never called and thus ```input``` is never fired.

**Comment by Dmatafonov** [comment]
> input event is triggered on successful upload only. If your server didn't return 200 status, uploadDone() function is never called and thus input is never fired.

That's the case. Should be triggered twice I think, because html actually changed twice - before and after successfull upload.

**Comment by orthes** [comment]
The problem is that when you trigger input before upload is complete, html contains raw preview images: ```<img src="data:image/jpeg;base64,...```. And if you're uploading a few MB of images, your html will have a few MB itself until upload completes. 

So if you're autosaving your content on input event, it's better to wait until upload completes. That's why input is not triggered before that.

Do you have a use case where you need input event sooner?

**Comment by Dmatafonov** [comment]
Yes, I do have such usecase.

In input event I save his blog to localstorage, so user doesn't loose his work event if browser refreshes.
If for some case server can't save image at this point (e.g) user session is expired, his work will be saved, including images.

It's not a problem to decode base64 when user actually submits the post, (actually I do it for ctrl+v images paste), but it is a problem that he does sees the image in post but it's not saved (because 'input' was not triggered).

**Comment by Dmatafonov** [comment]
Of course there can be some hacks but why not trigger input event when html actually had been changed? That's what input event is for. 

If he uploads image successfully, the source of file will be changed and input will be triggered again. And in that case some other custom events should be triggered ( e.g. 'mediumEditor_insert_uploadSuccessfull' or 'mediumEditor_insert_uploadError').

**Comment by wkurosawa** [comment]
I'm not sure if this should have been solved with #167 issue, or if it's another scenario, but I'm using MediumEditor 5.10, on a textarea input and when I insert an image it's not triggering `on input` event. 

I noticed this running some scenarios:
- Input only text -> OK -> POST data is a HTML with paragraphs elements
- Input text, insert image and after upload input more text -> OK -> POST data is a HTML with paragraphs and the image element
- Input text, insert image -> Fail -> POST data is a HTML only with paragraph and no image.
- Insert only image -> Fail -> POST data is empty

My app have an uploader and it's everything ok on successful scenarios (1 and 2): it really uploads images and works like a charm. 
But if you upload an image and do nothing after, it doesn't trigger MediumEditor's `handleInput` that should update input hidden field. 
App: https://medium-insert-plugin-example.herokuapp.com/
Code: https://github.com/wkurosawa/medium-insert-plugin-example

It's an example HTML. Even though it's not really uploading, and gives `upload.php` error, it should have been able to update textarea POST with base64 image.  
![me-meip](https://cloud.githubusercontent.com/assets/1878698/11191989/ae4a9596-8c85-11e5-89ef-cebb38033d3d.gif)
